client/lib/signup/progress-store.js

```diff
@@ -4,139 +4,85 @@
  * External dependencies
  */
 
-import { assign, clone, filter, find, get, isEmpty, map, omit } from 'lodash';
+import { cloneDeep, isEqual, get, isEmpty, omit } from 'lodash';
 import debugFactory from 'debug';
 const debug = debugFactory( 'calypso:signup-progress-store' );
-import store from 'store';
 
 /**
  * Internal dependencies
  */
 import Dispatcher from 'dispatcher';
-import emitter from 'lib/mixins/emitter';
-import SignupDependencyStore from './dependency-store';
-import flows from 'signup/config/flows';
 import steps from 'signup/config/steps-pure';
-
-/**
- * Constants
- */
-const STORAGE_KEY = 'signupProgress';
-
-/**
- * Module variables
- */
-let signupProgress = [];
+import { SIGNUP_COMPLETE_RESET, SIGNUP_PROGRESS_SET } from 'state/action-types';
+import {
+	completeStep,
+	processStep,
+	removeUnneededSteps,
+	saveStep,
+	invalidateStep,
+	submitStep,
+} from 'state/signup/progress/actions';
+import { getSignupProgress } from 'state/signup/progress/selectors';
+import SignupDependencyStore from './dependency-store';
 
 const SignupProgressStore = {
-	get: function() {
-		return clone( signupProgress );
+	currentValue: null,
+	reduxStore: null,
+	subscribers: new Map(),
+
+	get() {
+		if ( ! this.reduxStore ) {
+			throw new Error( 'Tried to read redux state before store was ready' );
+		}
+		return cloneDeep( getSignupProgress( this.reduxStore.getState() ) );
```

Comment:
why the `cloneDeep`?

Reply:
it was more curiosity than anything. I assume this code is going to get thrown out anyway when we remove the flux store, so ... probably fine to leave it. `cloneDeep` isn't cheap.

Reply:
I just reviewed all instances of `ProgressStore.get()` and confirmed that we don't mutate the object anymore with this change. I'll go ahead and remove it.